src/app/current-users/current-users.component.ts

```diff
@@ -12,64 +13,73 @@ import { UsersObject } from '../../../api-objects/UsersObject';
+      .subscribe(
+        (results) => {
+          this.usersTable = results.data;
+        });
+  }
 
-  this.usersTable = Users;
-  // this.usersTable.map((types) => {
-  //   if(types.authorityType === 0) {} else {}
-  // });
- }
+  public prevPage(pageNum: number): void {
+    while (this.pageCounter > 0 && pageNum > 0) {
+      this.pageCounter -= 1;
+      pageNum -= 1;
+    }
+  }
 
- public prevPage(pageNum: number): void {
-   while( this.pageCounter > 0 && pageNum > 0) {
-     this.pageCounter -= 1;
-     pageNum -= 1;
-   }
- }
+  public nextPage(pageNum: number): void {
+    while ((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
+      this.pageCounter += 1;
+      pageNum += 1;
+    }
+  }
 
- public nextPage(pageNum: number): void {
-   while((this.pageCounter + 1) * 10 < this.usersTable.length && pageNum > 0) {
-     this.pageCounter += 1;
-     pageNum += 1;
-   }
- }
+  /**
+   * Deletes a user at a given index.
+   * @param user a user object to be deleted
+   */
+  public deleteUser(user: UsersObject) {
+    this.apiUsersCrudService.deleteUser(user.username).subscribe(
+      (_) => {
+        let index: number = this.usersTable.indexOf(user);
+        this.usersTable.splice(index, 1);
+        this.notificationService.showSuccess('User '+ user.username + ' was successfully deleted');
+      }
+    )
+  }
 
- /**
-  * Deletes a user at a given index.
-  * @param user a user object to be deleted
-  */
- public deleteUser(user: UsersObject) {
-   let index: number = this.usersTable.indexOf(user);
-   this.usersTable.splice(index, 1);
- }
+  /**
+   * Updates the username of the user, at a given index.
+   * @param f NgForm that contains the value inside the form tag in the template view
+   * @param update UsersObject that keeps track of the index.
+   */
+  public updateUser(f: NgForm, update: UsersObject) {
+    let index: number = this.usersTable.indexOf(update);
+    if (f.value.username !== '') {
+      this.apiUsersCrudService.updateUsername(update.username, f.value.username).subscribe(
+        (_) => {
+          this.notificationService.showSuccess('Username, ' + update.username + ' was successfully updated to ' + f.value.username);
+          this.usersTable[index].username = f.value.username;
+        }
+      )
+    }
+  }
 
- /**
-  * Updates the username of the user, at a given index.
-  * @param f NgForm that contains the value inside the form tag in the template view
-  * @param update UsersObject that keeps track of the index.
-  */
- public updateUser(f: NgForm , update: UsersObject) {
-   let index: number = this.usersTable.indexOf(update);
-   if(f.value.username !== '') {
-     this.usersTable[index].username = f.value.username;
-   }
- }
+  public resetPassword(user: UsersObject) {
+    this.apiUsersCrudService.resetPassword(user.username).subscribe(
+      result => {
+        this.notificationService.showSuccess('NEW PASSWORD FOR ' + user.username + ' IS: ' + result.data);
```

Comment:
This is not a great way to reveal the password because it quickly disappears.

It would be better to create a modal dialogue that can later also be used to enter your password for confirmation.